client/src/game/index.ts

```diff
@@ -0,0 +1,181 @@
+import { Material, Mesh, Object3D, PerspectiveCamera, Scene } from 'three'
+import { v4 as uuid } from 'uuid'
+import ModelLoader, { LoadedGameModel } from './loaders/modelLoader'
+import GameObject, { IGameObject } from './gameObjects/gameObject'
+import MoveableGameObject from './gameObjects/moveableGameObject'
+import GameStateDataService from './services/gameStateDataService'
+import { CreateUnitMessage, SetUnitMessage } from './services/models'
+import GameWorld from './gameWorld'
+import GameRenderer from './gameRenderer'
+import GameControls from './controls/gameControls'
+
+class Game {
+	private scene: Scene
+
+	private camera: PerspectiveCamera
+
+	private gameRenderer: GameRenderer
+
+	private gameControls: GameControls
+
+	private modelLoader: ModelLoader
+
+	private gameWorld: GameWorld
+
+	private gameStateDataService: GameStateDataService
+
+	constructor(container: HTMLDivElement) {
+		this.scene = new Scene()
+		this.camera = new PerspectiveCamera(75, window.innerWidth / window.innerHeight, 0.1, 200)
+
+		this.gameWorld = new GameWorld(this.scene)
+		this.gameRenderer = new GameRenderer(container, this.scene, this.camera)
+		this.gameControls = new GameControls(container, this.camera, this.gameWorld)
+		this.modelLoader = new ModelLoader()
+		this.gameStateDataService = new GameStateDataService()
+	}
+
+	public initialize = (): void => {
+		this.gameRenderer.initialize()
+		this.gameControls.initialize()
+		this.modelLoader.initialize()
+		this.gameWorld.initialize()
+
+		this.gameRenderer.addUpdateTarget(this.gameWorld)
+		this.gameRenderer.addUpdateTarget(this.gameControls)
+
+		this.camera.position.set(30, 30, -20)
+		this.loadModels()
+	}
+
+	private connectToServer = (connectionAtempt = 0): void => {
+		this.gameStateDataService
+			.connectToWebsocket(1)
+			.then((connected) => this.handleConnectedToServer(connected, connectionAtempt + 1))
+			.catch(() => this.handleConnectedToServer(false, connectionAtempt + 1))
+	}
+
+	private handleConnectedToServer = (connected: boolean, connectionAtempt = 0): void => {
+		if (connected) {
+			this.gameStateDataService.addMessageHandler('CreateUnit', this.handleServerCreateUnit)
+			this.gameStateDataService.addMessageHandler('SetUnit', this.handleServerSetUnit)
+
+			this.gameRenderer.start()
+		} else {
+			setTimeout(() => {
+				this.connectToServer(connectionAtempt)
+			}, 5000 * connectionAtempt)
+		}
+	}
+
+	private handleServerCreateUnit = (message: CreateUnitMessage): void => {
+		// eslint-disable-next-line no-console
+		console.log('handleServerCreateUnit', message)
```

Comment:
Shouldn't we actually create a unit here instead?

Reply:
yes i just wanted to get my changes in before we had more merge conflicts
im working on it on another branch